### PR TITLE
Eslint Cleanup

### PR DESCRIPTION
**Pull request by englercj**
Added a more specific and restrictive eslint file, and ran `--fix` on source and tests.

Includes some good stuff like:

- consistent spacing
- removing trailing newlines
- enforcing a newline at end of file
- use template strings instead of concatenation
- enforcing consistent comment formats
- validates jsdoc
- ensures there is always a trailing comma for multi-line objects/arrays (helps cleanup diffs)

Still more to do until there are no errors, but this is a start. I would like to fix as much of this as is possible *before* I branch the `next` branch to minimize conflicts.

When I left off, here are the sections i had left to do:

- [x] core
- [x] extract
- [x] extras
- [x] filters
- [x] interaction
- [x] loaders
- [x] mesh
- [x] particles

### CODE REVIEW

**Comment by bigtimebuddy** [comment]
@englercj You're insane for attempting this! but I'm totally behind it. Let me know if you want any help.

**Comment by stbaer** [comment]
Thats a lot :) Same here, would also contribute sth. But not sure how to contribute to a pr.

**Comment by englercj** [comment]
OK, linting passes, tests pass, and I squashed into a single commit. Phew!

File: src/core/const.js [hunk 1]
```diff
@@ -1,400 +1,420 @@
...
+ * @property {number} DEFAULT='mediump'
+ * @property {number} LOW='lowp'
+ * @property {number} MEDIUM='mediump'
+ * @property {number} HIGH='highp'
+ */
+export const PRECISION = {
+    DEFAULT: 'mediump',
+    LOW: 'lowp',
+    MEDIUM: 'mediump',
+    HIGH: 'highp',
```
Comment (bigtimebuddy) [inline src/core/const.js]: Why are the values sometimes white spaced aligned (e.g, `TRANSFORM_MODE` below) and some are not, like this? Seems inconsistent. Also some white-spaced value aligned objects have  have a single space and some have lots of spacing.

File: src/core/display/TransformStatic.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import math from '../math';
+import * as math from '../math';
```
Comment (bigtimebuddy) [inline src/core/display/TransformStatic.js]: `import {ObservablePoint} from '../math';`?

File: src/core/graphics/webgl/shaders/PrimitiveShader.js [hunk 3]
```diff
@@ -29,18 +31,16 @@ class PrimitiveShader extends Shader
                 'void main(void){',
                 '   gl_Position = vec4((projectionMatrix * translationMatrix * vec3(aVertexPosition, 1.0)).xy, 0.0, 1.0);',
                 '   vColor = aColor * vec4(tint * alpha, alpha);',
-                '}'
+                '}',
```
Comment (bigtimebuddy) [inline src/core/graphics/webgl/shaders/PrimitiveShader.js]: Why trailing comma? Seems like a weird stylistic choice.

**Comment by bigtimebuddy** [comment]
@englercj epic PR! Overall, great work and brings a lot of sanity to the code style and conforms everything. Love that it's ESLint enforced. Reworking of the imports and exports is nice, especially when importing CONSTS. I didn't get to go through all of it, but left a few comments/questions.

**Comment by GoodBoyDigital** [comment]
👍

File: src/core/display/TransformStatic.js (hunk 2 above)
Comment (englercj) [inline src/core/display/TransformStatic.js]: Yeah we need to do a pass where we change all the `import * as` into only the imports needed. Just didn't want to hold up this entire PR for that pass.

File: src/core/const.js (hunk 1 above)
Comment (englercj) [inline src/core/const.js]: Some where aligned and some weren't before, just didn't change it.

File: src/core/graphics/webgl/shaders/PrimitiveShader.js (hunk 3 above)
Comment (englercj) [inline src/core/graphics/webgl/shaders/PrimitiveShader.js]: Cleans up diffs. If you add/change a line it doesn't have changes elsewhere in the diff. By always having a trailing comma you ensure your diffs remain very clean and obvious as to what has changed. Then we can let the transpiler deal with removing it if necessary for compat.

**Comment by englercj** [comment]
Rebased to include the changes in dev.

**Pull request merged**